src/ide/ide.ts

```diff
@@ -820,13 +823,97 @@ export let ide = (function () {
 				module.interpreter.flush();
 			};
 			module.interpreter.service.alert = function (msg) {
-				alert(msg);
+				return new Promise((resolve, reject) => {
+					let dlg = tgui.msgBox({
+						title: "",
+						prompt: msg,
+						icon: tgui.msgBoxExclamation,
+						buttons: [{ text: "Close", isDefault: true }],
```

Comment:
```suggestion
						buttons: [{ text: "Okay", isDefault: true }],
```

I think I'd rather prefer it without an icon and with a different button label, since that's how most browsers show alerts 🤷‍♂️